src/musicPlaylistSecondary.java

```diff
@@ -0,0 +1,409 @@
+import java.util.Random;
+
+public abstract class musicPlaylistSecondary implements musicPlaylist {
+
+    /**
+     * Private helper: reports whether this playlist is empty by checking the
+     * kernel method getCurrentSong().
+     *
+     * @return true if the playlist is empty, false otherwise
+     */
+    private boolean isEmpty() {
+        // If getCurrentSong() returns the default "No songs in playlist",
+        // we consider it empty.
+        return "No songs in playlist".equals(this.getCurrentSong());
```

Comment:
Very interesting design. Personally, I would have a length method or something like it. Then I would have a @requires in `getCurrentSong()` that makes it so you can't call it if the playlist is empty.